### DISCUSSION TITLE
How to reload tracing-sampling-settings.yml without restart OAP server

### DISCUSSION THREAD

**Discussion by xiaozongyang** [apache/skywalking#12695] [Q&A] [answered]
Is there a way to trigger OAP server reload `tracing-sampling-settings.yml` without restarting? Just like `kill -HUP <pid-of-prometheus>` or `curl -XPOST <prometheus-ip-and-port>/-/reload` does.

I try to send a `HUP` sginal to OAP server process, the proces quited.

**Comment by xiaozongyang** [comment]
ping @wu-sheng

**Comment by wu-sheng** [comment, marked as answer]
There is no reload, because it is not elegance.

Dynamic configuration is loading new configuration from another server, rather than reloading from fs.

**Reply by wu-sheng** [reply]
Use agent-analyzer.default.traceSamplingPolicy through https://skywalking.apache.org/docs/main/latest/en/setup/backend/dynamic-config/

**Comment by xiaozongyang** [comment]
@wu-sheng Thanks a lot!
So if I want to use dynamic configuration feature, I must integrate a third-party storage in list of [dynamic-configuration-implementations](https://skywalking.apache.org/docs/main/latest/en/setup/backend/dynamic-config/#dynamic-configuration-implementations). 
If I put the raw `tracing-sampling-policy-settings.yml` in a configmap and mount it for the OAP pod in k8s, that is not working for dynamic configuration purpose.
Do I have any mis-understandings?

**Reply by wu-sheng** [reply]
I don't follow what do you mean mount. It is a SW_CLUSTER_K8S_LABEL driven watching for k8s config map. Not load a file to oap pod.
